unity/CAVE/Assets/W3D_Translator/Classes.cs

```diff
@@ -0,0 +1,1519 @@
+        [XmlElement(ElementName="None")]
+        public object None;
+
+        [XmlElement(ElementName="Text")]
+        public Text Text;
+
+        [XmlElement(ElementName="Image")]
+        public Image Image;
+
+        [XmlElement(ElementName="StereoImage")]
+        public StereoImage StereoImage;
+
+        // TODO: Model
+        [XmlElement(ElementName="Model")]
+        public Model Model;
+
+        [XmlElement(ElementName="Light")]
+        public Light Light;
+
+        [XmlElement(ElementName="ParticleSystem")]
+        public ParticleSystem ParticleSystem;
+    }
+
+
+    [XmlRoot(ElementName="Text")]
+    public class Text
+    {
+
+        [XmlElement(ElementName="text")]
+        public List<string> text;
+
+        // TODO: Enum
+        [XmlAttribute(AttributeName="horiz-align")]
+        public string horizontalAlign;
+
+        // TODO: Enum
+        [XmlAttribute(AttributeName="vert-align")]
+        public string verticalAlign;
+
+        [XmlAttribute(AttributeName="font")]
+        public string font;
+
+        [XmlAttribute(AttributeName="depth")]
+        public double depth;
+    }
+
+
+    [XmlRoot(ElementName="Image")]
+    public class Image
+    {
+        // TODO: file simple type
+        [XmlAttribute(AttributeName="filename")]
+        public string filename;
+    }
+
+    [XmlRoot(ElementName="StereoImage")]
+    public class StereoImage
+    {
+        // TODO: file simple type
+        [XmlAttribute(AttributeName="left-image")]
+        public string leftImage;
+
+        // TODO: file simple type
+        [XmlAttribute(AttributeName="right-image")]
+        public string rightImage;
+    }
+
+    [XmlRoot(ElementName="Model")]
+    public class Model
+    {
+        // TODO: file simple type
+        [XmlAttribute(AttributeName="filename")]
+        public string filename;
+
+        [XmlAttribute(AttributeName="check-collisions")]
+        public string checkCollisions;
+    }
+
+
+    [XmlRoot(ElementName="ParticleSystem")]
+    public class ParticleSystem
+    {
+        [XmlAttribute(AttributeName="max-particles")]
+        public int maxParticles;
+
+        [XmlAttribute(AttributeName="actions-name")]
+        public string actionsName;
+
+        [XmlAttribute(AttributeName="particle-group")]
+        public string particleGroup;
+
+        [XmlAttribute(AttributeName="look-at-camera")]
+        public bool lookAtCamera;
+
+        [XmlAttribute(AttributeName="sequential")]
+        public bool sequential;
+
+        [XmlAttribute(AttributeName="speed")]
+        public double speed;
+    }
+
+
+    // Content.Light
+
+
+    [XmlRoot(ElementName="Light")]
+    public class Light
+    {
+        // TODO: Enum (Point, Directional, Spot)
+        [XmlElement(ElementName="Point")]
+        public object Point;
+
+        [XmlElement(ElementName="Directional")]
+        public object Directional;
+
+        [XmlElement(ElementName="Spot")]
+        public object Spot;
```

Comment:
How do the `object`s work here? Is C# smart enough to autopopulate their data members based on the XML schema or something? Can you access the data members of the a `Directional` light without defining a `DirectionalLight` class?

Reply:
These are leaves so they don't really hold any data, nor do they have a type. What's really happening is a `<Light>` will either have `<Directional />`, `<Point />`, or `<Spot />` as a child. 

There's an `[XmlChoiceIdentifier()]` attribute that's SUPER powerful in these cases which I add in an upcoming PR. It basically becomes a single `enum`, then I can run a `switch` block in `CLI.cs`. Otherwise the tool that generated the code was smart enough to create a new class to hold any nested information.

```xml
<Light diffuse="true" specular="false" const_atten="1.0" lin_atten="0.0" quad_atten="0.0">
    <Directional/>
</Light>
```